core/src/banking_stage/transaction_scheduler/scheduler_controller.rs

```diff
@@ -143,6 +149,53 @@ impl SchedulerController {
         }
     }
 
+    /// Clean unprocessable transactions from the queue. These will be transactions that are
+    /// expired, already processed, or are no longer sanitizable.
+    /// This only clears pending transactions, and does **not** clear in-flight transactions.
+    fn clean_queue(&mut self) {
+        // Clean up any transactions that have already been processed, are too old, or do not have
+        // valid nonce accounts.
+        const MAX_TRANSACTION_CHECKS: usize = 10_000;
+        let mut transaction_ids = Vec::with_capacity(MAX_TRANSACTION_CHECKS);
+
+        while let Some(id) = self.container.pop() {
```

Comment:
Yeah, it could be. My intent here was to keep the runtime of this fn very short by only cleaning the top-of-queue, i.e. only trnasactions we're very likely to schedule early on during leader slot.